common/peerprovider/ringpopprovider/provider.go

```diff
@@ -171,13 +186,53 @@ func (r *Provider) SelfEvict() error {
 }
 
 // GetMembers returns all hosts with a specified role value
-func (r *Provider) GetMembers(service string) ([]string, error) {
-	return r.ringpop.GetReachableMembers(swim.MemberWithLabelAndValue(roleKey, service))
+func (r *Provider) GetMembers(service string) ([]*membership.HostInfo, error) {
+	var res []*membership.HostInfo
+
+	// filter member by service name, add port info to Hostinfo if they are present
+	memberData := func(member swim.Member) bool {
+		portMap := make(membership.PortMap)
+		if v, ok := member.Label(roleKey); !ok || v != service {
+			return false
+		}
+
+		if v, ok := member.Label(portTchannel); ok {
+			port, err := labelToPort(v)
+			if err != nil {
+				r.logger.Warn("port cannot be converted", tag.Error(err), tag.Value(v))
```

Comment:
Maybe `tchannel port cannot be converted` and `grpc port cannot be converted` below since there two identical warnings currently, it may not be obvious which one is logged.